### DISCUSSION TITLE
Set Preprocessor Macros

### DISCUSSION THREAD

**Discussion by anon767** [Fraunhofer-AISEC/cpg#613] [Q&A] [answered]
Hi,
is there currently a way to set C Preprocessor macros or to entirely ignore them?
The motivation is, that I want to analyze larger projects and some use macros i.e.:

```
#ifdef FEATURE_XY_ENABLED
 [... stuff ...]
#end if
```

The result is of course an empty AST :( 

Thanks in advance

**Comment by oxisto** [comment, marked as answer]
Yes, using the `symbols` parameter of a `TranslationConfiguration`.

https://github.com/Fraunhofer-AISEC/cpg/blob/ba1ecaa8397e993e44aa75f26486870b1a68cdbe/cpg-library/src/main/java/de/fraunhofer/aisec/cpg/TranslationConfiguration.java#L115
